internal/builder/builder_test.go

```diff
@@ -1757,6 +1757,63 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 			})
 		})
 	})
+
+	when("builder is flatten", func() {
+		var (
+			bldr         *builder.Builder
+			builderImage imgutil.Image
+		)
+
+		it.Before(func() {
+			h.AssertNil(t, baseImage.SetEnv("CNB_USER_ID", "1234"))
+			h.AssertNil(t, baseImage.SetEnv("CNB_GROUP_ID", "4321"))
+			h.AssertNil(t, baseImage.SetLabel("io.buildpacks.stack.id", "some.stack.id"))
+			h.AssertNil(t, baseImage.SetLabel("io.buildpacks.stack.mixins", `["mixinX", "mixinY", "build:mixinA"]`))
+			h.AssertNil(t, baseImage.SetLabel(
+				"io.buildpacks.builder.metadata",
+				`{"description": "some-description", "createdBy": {"name": "some-name", "version": "1.2.3"}, "buildpacks": [{"id": "buildpack-1-id"}, {"id": "buildpack-2-id"}], "groups": [{"buildpacks": [{"id": "buildpack-1-id", "version": "buildpack-1-version", "optional": false}, {"id": "buildpack-2-id", "version": "buildpack-2-version-1", "optional": true}]}], "stack": {"runImage": {"image": "prev/run", "mirrors": ["prev/mirror"]}}, "lifecycle": {"version": "6.6.6"}}`,
+			))
+			h.AssertNil(t, baseImage.SetLabel(
+				"io.buildpacks.buildpack.order",
+				`[{"group": [{"id": "buildpack-1-id", "optional": false}, {"id": "buildpack-2-id", "version": "buildpack-2-version-1", "optional": true}]}]`,
+			))
+
+			builderImage = baseImage
+		})
+
+		when("flatten all", func() {
```

Comment:
The only one could be "flatten with  depth", because "flatten none" is actually the other test cases that already exists which is the default behavior

Reply:
Makes sense. What about something like:
```	when("flatten", func() {
		when("all", func() {
		})
		when("with depth", func() {
		})
	})
```

Reply:
I will add a TODO, because the `with depth` test case requires to set a tree of composite buildpacks